### ISSUE TITLE
False positive: null reference

### ISSUE THREAD

**Issue by Bilge** [kalessil/phpinspectionsea#1874]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) 4.0.7.1 |
| Language level | PHP 8.1                        |

Current behaviour
--
```php
    private function myFunction(?object $obj): void
    {
        if ($obj?->foo()) {
            // [EA] Null pointer exception may occur here. 
            $obj->bar();
        }
    }
```

Expected behaviour
---
Given that `$obj?->foo()` must return `null` if `$obj` is `null`, the `if` clause thus guarantees `$obj` is not null within the guard clause. Ergo, this is a false positive.

**Comment by GlucNAc** [comment]
I have the same issue running 2022.3.3 and PHP Inspections EA extended 4.0.7.1. This problem is the same as https://github.com/kalessil/phpinspectionsea/issues/1826, and linked to https://github.com/kalessil/phpinspectionsea/issues/1861.

**Comment by asamofal** [comment]
I faced the same issue... PhpStorm 2023.1.1 and  PHP Inspections (EA Extended) v 5.0.0.0

```php
$payment = $reservation->getPayment(); // it's nullable
if (
    $payment?->type === Reservation::PAYMENT_TYPE_KLARNA
    && $payment?->state === KlarnaOrder::STATUS_AUTHORIZED
) {
    $klarnaOrder = new KlarnaOrder($payment->id); // [EA] Null pointer exception may occur here. 
}
```
